src/autoscaler/api/server/api_handler.go

```diff
@@ -131,14 +141,51 @@ func (h *ApiHandler) BindServiceInstance(w http.ResponseWriter, r *http.Request,
 		return
 	}
 
-	if instanceId == "" || bindingId == "" || body.ServiceID == "" || body.PlanID == "" {
+	if body.AppID == "" || instanceId == "" || bindingId == "" || body.ServiceID == "" || body.PlanID == "" {
 		handlers.WriteJSONResponse(w, http.StatusBadRequest, models.ErrorResponse{
 			Code:    "Bad Request",
 			Message: "Malformed or missing mandatory data",
 		})
 		return
 	}
 
+	if body.Policy == "" {
+		h.logger.Info("no policy json provided", lager.Data{})
+	} else {
+		errResults, valid := h.policyValidator.ValidatePolicy(body.Policy)
+
+		if !valid {
+			handlers.WriteJSONResponse(w, http.StatusBadRequest, errResults)
+			return
+		}
+
+		policyGuid, err := uuid.NewV4()
+		if err != nil {
+			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+				Code:    "Interal-Server-Error",
+				Message: "Error generating policy guid"})
+			return
+		}
+
+		h.logger.Info("saving policy json", lager.Data{"policy": body.Policy})
+		err = h.policydb.SaveAppPolicy(body.AppID, body.Policy, policyGuid.String())
+		if err != nil {
+			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+				Code:    "Interal-Server-Error",
+				Message: "Error saving policy"})
+			return
+		}
+
+		h.logger.Info("creating/updating schedules", lager.Data{"policy": body.Policy})
+		err = h.schedulerUtil.CreateOrUpdateSchedule(body.AppID, body.Policy, policyGuid.String())
+		if err != nil {
+			handlers.WriteJSONResponse(w, http.StatusInternalServerError, models.ErrorResponse{
+				Code:    "Interal-Server-Error",
+				Message: "Error creating/updating schedules"})
+			return
+		}
+	}
+
 	err = h.bindingdb.CreateServiceBinding(bindingId, instanceId, body.AppID)
```

Comment:
Referring to the original service broker implementation, we need to create service binding prior to the policy handling part. 
https://github.com/cloudfoundry-incubator/app-autoscaler/blob/master/servicebroker/lib/routes/binding.js#L63-L130